client/lib/products-list/index.js

```diff
@@ -100,15 +101,13 @@ ProductsList.prototype.initialize = function( productsList ) {
 };
 
 /**
- * Parses data returned from the API.
+ * Parses data retrieved from the API and extracts the list of products.
  *
- * @param {array} data
- * @return {array}
+ * @param {object} data - raw data
+ * @return {object} a list of products
  **/
 ProductsList.prototype.parse = function( data ) {
-	delete data._headers;
-
-	return data;
+	return omit( data, '_headers' );
```

Comment:
Unlike features/countries/plans-list, the products-list endpoint returns an object, so _.omit is appropriate here. :+1: